### ISSUE TITLE
/pa spectate

### ISSUE THREAD

**Issue by falget** [slipcor/pvparena#319]
If the arena is active and user will write the command /pa spectate then write /pa leave, after that the current player and all players in arena will be kick from arena.

**Comment by Oruss7** [comment]
can you join the version of pvpArena please ?

**Comment by falget** [comment]
Version 1.3.4.287

**Comment by Oruss7** [comment]
Can't reproduce. Can you add your arena config please ?

**Comment by falget** [comment]
[sarena.txt](https://github.com/slipcor/pvparena/files/1223125/sarena.txt)
Also i use mods pa_m_duel.

**Comment by slipcor** [comment]
The linking of your arena does not seem to have worked

Sorry for the delay, the notifications did not inform me of updates :O

What exactly are people told when they are being kicked, are there warnings or errors in the console?

**Comment by falget** [comment]
People kicked as if as one of them wrote /pa l, and duel end. In the console only few message:
> [12:56:18] [Server thread/INFO]: [PVP Arena] Error: No team found!
> [12:56:18] [Server thread/INFO]: [PVP Arena] Error: No team found!
> [12:56:19] [Server thread/INFO]: [PVP Arena] Error: No team found!
but this message also show when battle in the arena ended.

**Comment by slipcor** [comment]
Do players get a message like "you left the fight" or anything or do they just leave?

**Comment by falget** [comment]
In chat say to current player that he leave from pvp arena and say player which leave from spectate, leave from the fight.

**Comment by slipcor** [comment]
ok so is this what happens?

- an arena match is going on
- someone joins with /pa spectate
- this person then uses /pa leave

result: everyone (spectator and fighters) gets kicked with message "you left the fight" ?

**Comment by falget** [comment]
Yes but, spectator see message: "you left the arena" and fighters see messages: "player(spectator name) left the fight" and "you left the arena"

**Comment by slipcor** [comment]
yes, this is what I meant! Okay, I will look into it....

**Comment by slipcor** [comment]
Check out the latest duel module version, I fixed this issue. Thanks for reporting - this was really hard to track down because you need 3 people - but nothing you could have done would have showed it, I had to add stacktraces to find what was going on ^^

https://ci2.craftyn.com/job/PVP%20Arena%20Modules/
